extensions/amp-animation/amp-animation.md

```diff
@@ -25,278 +25,92 @@ limitations under the License.
-    <td><code>direction</code></td>
-    <td>string</td>
-    <td>"normal" </td>
-    <td>One of "normal", "reverse", "alternate" or "alternate-reverse".</td>
-  </tr>
-  <tr>
-    <td><code>fill</code></td>
-    <td>string</td>
-    <td>"none"</td>
-    <td>One of "none", "forwards", "backwards", "both", "auto".</td>
-  </tr>
-</table>
-
-All timing properties allow either a direct numeric/string values or CSS values. For instance, "duration" can be specified as `1000` or `1s` or `1000ms`. In addition, `calc()` and `var()` and other CSS expressions are also allowed.
-
-An example of timing properties in JSON:
-
-```text
-{
-  ...
-  "duration": "1s",
-  "delay": 100,
-  "endDelay": "var(--end-delay, 10ms)",
-  "easing": "ease-in",
-  "fill": "both"
-  ...
-}
+```html
+<amp-animation layout="nodisplay">
+  <script type="application/json">
+    [
+      {
+        "selector": ".target-class",
+        "duration": 1000,
+        "keyframes": {"opacity": 1}
+      },
+      {
+        "selector": ".target-class",
+        "duration": 600,
+        "delay": 400,
+        "keyframes": {"transform": "scale(2)"}
+      }
+    ]
+  </script>
+</amp-animation>
 ```
 
-Animation components inherit timing properties specified for the top-level animation.
+Trigger animation start via the `trigger` attribute or an [action](#actions).
 
-### Subtargets
+You may place `amp-animation` controlled via actions anywhere in the DOM. If the animation contains `trigger="visibility"` it must be a direct child of the `<body>`.
 
-Everywhere where `selector` can be specified, it's possible to also specify `subtargets: []`. Subtargets can override timing properties or variables defined in the animation for specific subtargets indicated via either an index or a CSS selector.
+### Defining effects
 
-For instance:
+#### Keyframes
 
-```text
-{
-  "selector": ".target",
-  "delay": 100,
-  "--y": "100px",
-  "subtargets": [
-    {
-      "index": 0,
-      "delay": 200,
-    },
-    {
-      "selector": ":nth-child(2n+1)",
-      "--y": "200px"
-    }
-  ]
-}
-```
-
-In this example, by default all targets matched by the ".target" have delay of 100ms and "--y" of 100px. However, the first target (`index: 0`) is overriden to have delay of 200ms; and odd targets are overriden to have "--y" of 200px.
-
-Notice, that multiple subtargets can match one target element.
-
-### Keyframes
+You must declare effects as keyframes to apply animations using `amp-animations`.
 
-Keyframes can be specified in numerous ways described in the [keyframes section](https://www.w3.org/TR/web-animations/#processing-a-keyframes-argument) of the Web Animations spec or as a string refering to the `@keyframes` name in the CSS.
+You may specify keyframes in amp-animation in the same way as defined in MDN's [Keyframe Formats](https://developer.mozilla.org/en-US/docs/Web/API/Web_Animations_API/Keyframe_Formats). You may also reference the @keyframes name defined as CSS within the `<style amp-custom>` tag.
```

Comment:
`<style amp-custom>` or `<style amp-keyframes>` tags.

Reply:
Very good catch, thank you